### PR TITLE
add new cl_pocl_command_buffer extension APIs

### PR DESCRIPTION
**Pull request by franz**

### CODE REVIEW

File: lib/CL/clCommandReadBufferPOCL.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* OpenCL runtime library: clCommandReadBufferPOCL()
+
+   Copyright (c) 2022-2023 Michal Babej / Intel Finland Oy
```
Comment (pjaaskel) [inline lib/CL/clCommandReadBufferPOCL.c:3]: Also some other headers have an old year range.

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Thanks for dusting this off. Can you add a mention of these to the release notes?
Also, the *Rect functions do not seem to use the SVM cl_mem shadow buffer refactoring. I wonder do the implicit migrations work with PoCL-R?

**Comment by franz** [comment]
> Also, the *Rect functions do not seem to use the SVM cl_mem shadow buffer refactoring. 

Yes, this is a rebased & slightly cleaned up code from ~1 year ago, before the SVM cl_mem was merged.

> I wonder do the implicit migrations work with PoCL-R?

The ones you have changed should still work, these new ones do not, since they use the SVM pointer directly. But it should work with CPU devices. Also missing are tests; i did limited testing with chipStar graph PR when the code was written, but we should  have more tests (possibly in OpenCL-CTS).

**Comment by jansol** [comment]
#1433 should fix the RDMA CI

**Pull request merged**